networkx/algorithms/approximation/dominating_set.py

```diff
@@ -21,71 +21,96 @@
 #   Nicholas Mancuso <nick.mancuso@gmail.com>
 #   All rights reserved.
 #   BSD license.
-import networkx as nx
+from __future__ import division
+
+from ..matching import maximal_matching
+from ...utils import not_implemented_for
+
 __all__ = ["min_weighted_dominating_set",
            "min_edge_dominating_set"]
+
 __author__ = """Nicholas Mancuso (nick.mancuso@gmail.com)"""
 
 
+# TODO Why doesn't this algorithm work for directed graphs?
+@not_implemented_for('directed')
 def min_weighted_dominating_set(G, weight=None):
-    r"""Return minimum weight vertex dominating set.
+    """Returns a dominating set that approximates the minimum weight node
+    dominating set.
 
     Parameters
     ----------
     G : NetworkX graph
-      Undirected graph
+        Undirected graph.
 
-    weight : None or string, optional (default = None)
-        If None, every edge has weight/distance/weight 1. If a string, use this
-        edge attribute as the edge weight. Any edge attribute not present
-        defaults to 1.
+    weight : string
+        The node attribute storing the weight of an edge. If provided,
+        the node attribute with this key must be a number for each
+        node. If not provided, each node is assumed to have weight one.
 
     Returns
     -------
     min_weight_dominating_set : set
-      Returns a set of vertices whose weight sum is no more than log w(V) * OPT
+        A set of nodes, the sum of whose weights is no more than `(\log
+        w(V)) w(V^*)`, where `w(V)` denotes the sum of the weights of
+        each node in the graph and `w(V^*)` denotes the sum of the
+        weights of each node in the minimum weight dominating set.
 
     Notes
     -----
-    This algorithm computes an approximate minimum weighted dominating set
-    for the graph G. The upper-bound on the size of the solution is
-    log w(V) * OPT.  Runtime of the algorithm is `O(|E|)`.
+
+    This algorithm computes an approximate minimum weighted dominating
```

Comment:
I'm fine if we want to move to having a blank line after section headings, but we should do so consistently. Can you make this docstring self-consistent, or clue me in if there is a different rationale?